homeassistant/components/librehardwaremonitor/sensor.py

```diff
@@ -0,0 +1,108 @@
+"""Support for LibreHardwareMonitor Sensor Platform."""
+
+from __future__ import annotations
+
+from librehardwaremonitor_api.model import LibreHardwareMonitorSensorData
+
+from homeassistant.components.sensor import SensorEntity, SensorStateClass
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddConfigEntryEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from . import LibreHardwareMonitorCoordinator
+from .const import DOMAIN
+from .coordinator import LibreHardwareMonitorConfigEntry
+
+# Coordinator is used to centralize the data updates
+PARALLEL_UPDATES = 0
+
+STATE_MIN_VALUE = "min_value"
+STATE_MAX_VALUE = "max_value"
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: LibreHardwareMonitorConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Set up the LibreHardwareMonitor platform."""
+    lhm_coordinator: LibreHardwareMonitorCoordinator = config_entry.runtime_data
+
+    sensor_entities = [
+        LibreHardwareMonitorSensor(lhm_coordinator, sensor_data)
+        for sensor_data in lhm_coordinator.data.sensor_data.values()
+    ]
+
+    async_add_entities(sensor_entities)
+
+
+class LibreHardwareMonitorSensor(
+    CoordinatorEntity[LibreHardwareMonitorCoordinator], SensorEntity
+):
+    """Sensor to display information from LibreHardwareMonitor."""
+
+    _attr_state_class = SensorStateClass.MEASUREMENT
+    _attr_has_entity_name = True
+
+    def __init__(
+        self,
+        coordinator: LibreHardwareMonitorCoordinator,
+        sensor_data: LibreHardwareMonitorSensorData,
+    ) -> None:
+        """Initialize an LibreHardwareMonitor sensor."""
+        super().__init__(coordinator)
+
+        self._attr_name: str = sensor_data.name
+        self.value: str | None = sensor_data.value
+        self.attributes: dict[str, str] = {
+            STATE_MIN_VALUE: self._format_number_value(sensor_data.min),
+            STATE_MAX_VALUE: self._format_number_value(sensor_data.max),
+        }
+        self._unit_of_measurement: str | None = sensor_data.unit
+        self._attr_unique_id: str = f"lhm-{sensor_data.sensor_id}"
+
+        self._sensor_id: str = sensor_data.sensor_id
+
+        # Hardware device
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, sensor_data.device_name)},
```

Comment:
What is the device name in this case? Is it truly unique?

Reply:
The device name is the actual hardware device the sensor belongs to. So for example:
- sensor 1: "temperature-gpu-core"
- sensor 2: "temperature-gpu-hotspot"
- sensor 3: "load-gpu-core"
will all share the same device name, e.g. something like "NVIDIA RTX 4080".
They will be grouped accordingly in the UI.

Reply:
Do we have anything that looks like a unique identifier?

Reply:
Actually there is a unique identifier. I have adapted the library to expose it and made the necessary changes to the integration's code. It should be fine now, even if someone has the same SSD twice in their PC.